### ISSUE TITLE
Implement useless try-except detection (`TRY302`)

### ISSUE THREAD

**Issue by dorschw** [charliermarsh/ruff#4308]
**Bad code:**
```python
# case 1
try:
    do_thing()
except Exception:
    raise
```

```python
# case 2
try:
    do_thing()
except:
    raise
```

```python
# case 3
try:
    do_thing()
except SomeKindOfException:
    raise
```

**Fix:**
remove the try/except, as it has no effect. 
```python
do_thing()
```
In cases 1&2, all exceptions are raised immediately, and in case 3 only a subset of exceptions is caught, then reraised immediately. 

Notes: 
* This shouldn't run if there's any other code in the `except` block. 
* If there are multiple excpt cases, empty ones (with only `raise`) should be removed, and others should remain.

**Comment by Skylion007** [comment]
This feels like a rule https://github.com/guilatrova/tryceratops/ would love to implement. @dorschw would you mind opening an issue there. If they implement it, we can implement it in ruff and use the same error code.

**Comment by JonathanPlasse** [comment]
It has been implemented by `tryceratops`.
It would be a good first issue.
